### PR TITLE
feat: make the email client explicity set the format to be HTML

### PR DESCRIPTION
**Pull request by mansueli**
## What kind of change does this PR introduce?
This makes the emails to be explicitly set to be HTML formatted instead of plain text which helper older clients that cannot figure this out themselves. 

## What is the current behavior?

Email messages are sent as plain/text. 

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: internal/mailer/mailer.go [hunk 1]
```diff
@@ -29,8 +29,12 @@ type Mailer interface {
 func NewMailer(globalConfig *conf.GlobalConfiguration) Mailer {
 	mail := gomail.NewMessage()
 
-	// so that messages are not grouped under each other
-	mail.SetHeader("Message-ID", fmt.Sprintf("<%s@gotrue-mailer>", uuid.Must(uuid.NewV4()).String()))
+	mail.SetHeaders(map[string][]string{
+		// Make the emails explicitly set to be HTML formatted (to cover older email clients)
+		"Content-Type": {"text/html; charset=utf-8"},
```
Comment (J0) [inline internal/mailer/mailer.go:34]: Thanks for the contribution! There's an existing `SetBody` within the mailme package- https://github.com/supabase/mailme/blob/main/mailme.go#L71

Not sure if that will override the header. set here - if it doesn't then I think this should be good to go!

**Comment by gunawanarisandi** [comment]
Hmmm

File: internal/mailer/mailer.go (hunk 1 above)
Comment (mansueli) [reply internal/mailer/mailer.go:34]: It sets the body type there, but it doesn't make it clear in the header which confuses some old email clients. 
They are different/ won't override one another.

Comment (J0) [reply internal/mailer/mailer.go:34]: Cool, thanks! for the explanation

**Review by J0 — APPROVED** [review]

**Pull request merged**